server/jetstream.go

```diff
@@ -829,32 +854,83 @@ func (a *Account) JetStreamEnabled() bool {
 	return enabled
 }
 
-// Updates accounting on in use memory and storage.
+func (jsa *jsAccount) remoteUpdateUsage(sub *subscription, c *client, subject, _ string, msg []byte) {
+	const usageSize = 16
+
+	jsa.mu.Lock()
+	s := jsa.js.srv
+	if len(msg) != usageSize {
+		jsa.mu.Unlock()
+		s.Warnf("Received remote usage update that is wrong size: %d vs %d", len(msg), usageSize)
+		return
+	}
+	var rnode string
+	if li := strings.LastIndexByte(subject, btsep); li > 0 && li < len(subject) {
+		rnode = subject[li+1:]
+	}
+	if rnode == _EMPTY_ {
+		jsa.mu.Unlock()
+		s.Warnf("Received remote usage update with no remote node")
+		return
+	}
+	var le = binary.LittleEndian
+	memUsed, storeUsed := int64(le.Uint64(msg[0:])), int64(le.Uint64(msg[8:]))
+
+	if jsa.rusage == nil {
+		jsa.rusage = make(map[string]*jsaUsage)
+	}
+	// Update the usage for this remote.
+	if usage := jsa.rusage[rnode]; usage != nil {
+		// Decrement our old values.
+		jsa.memTotal -= usage.mem
+		jsa.storeTotal -= usage.store
+		usage.mem, usage.store = memUsed, storeUsed
+	} else {
+		jsa.rusage[rnode] = &jsaUsage{memUsed, storeUsed}
+	}
+	jsa.memTotal += memUsed
```

Comment:
If we don't exclude the current node above, are we not double accounting here with what is done in updateUsage()?

Reply:
See above.